services/provider/server/server.go

```diff
@@ -341,77 +345,314 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
+		kernelMountOptions[parts[0]] = parts[1]
+	}
+	rbdClientProfileName := consumerConfigMap.Data["rbd-client-profile"]
+	cephfsClientProfileName := consumerConfigMap.Data["cephfs-client-profile"]
+	radosNamespace := consumerConfigMap.Data["rados-namespace-name"]
+	svgName := consumerConfigMap.Data["svg-name"]
+	extR = append(extR,
+		&pb.ExternalResource{
+			Kind: "ClientProfile",
+			Name: rbdClientProfileName,
+			Data: mustMarshal(
+				&csiopv1a1.ClientProfileSpec{
+					Rbd: &csiopv1a1.RbdConfigSpec{
+						RadosNamespace: radosNamespace,
+					},
+					CephConnectionRef: corev1.LocalObjectReference{Name: consumerResource.Status.Client.Name},
+				},
+			),
+		},
+		&pb.ExternalResource{
+			Kind: "ClientProfile",
+			Name: cephfsClientProfileName,
+			Data: mustMarshal(
+				&csiopv1a1.ClientProfileSpec{
+					CephFs: &csiopv1a1.CephFsConfigSpec{
+						SubVolumeGroup:     svgName,
+						KernelMountOptions: kernelMountOptions,
+						RadosNamespace:     ptr.To(svgName),
+					},
+					CephConnectionRef: corev1.LocalObjectReference{Name: consumerResource.Status.Client.Name},
+				},
+			),
+		},
+	)
 
-	// Get mgr pod hostIP
-	podList := &corev1.PodList{}
-	listOpts := []client.ListOption{
-		client.InNamespace(s.namespace),
-		client.MatchingLabels(map[string]string{"app": "rook-ceph-mgr"}),
-	}
-	err = s.client.List(ctx, podList, listOpts...)
+	cephCluster, err := util.GetCephClusterInNamespace(ctx, s.client, s.namespace)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list pod with rook-ceph-mgr label. %v", err)
-	}
-	if len(podList.Items) == 0 {
-		return nil, fmt.Errorf("no pods available with rook-ceph-mgr label")
+		return nil, err
 	}
 
-	mgrPod := &podList.Items[0]
-	var port int32 = -1
+	// SID for RamenDR
+	rbdStorageID := calculateCephRbdStorageID(
+		cephCluster.Status.CephStatus.FSID,
+		radosNamespace,
+	)
+	rbdProvisionerSecretName := consumerConfigMap.Data["rbd-provisioner-secret"]
+	rbdNodeSecretName := consumerConfigMap.Data["rbd-node-secret"]
 
-	for i := range mgrPod.Spec.Containers {
-		container := &mgrPod.Spec.Containers[i]
-		if container.Name == "mgr" {
-			for j := range container.Ports {
-				if container.Ports[j].Name == "http-metrics" {
-					port = container.Ports[j].ContainerPort
-				}
+	// SID for RamenDR
+	cephFsStorageID := calculateCephFsStorageID(
+		cephCluster.Status.CephStatus.FSID,
+		svgName,
+	)
+	cephFsProvisionerSecretName := consumerConfigMap.Data["cephFs-provisioner-secret"]
+	cephFsNodeSecretName := consumerConfigMap.Data["cephFs-node-secret"]
+
+	for i := 0; i < len(consumerResource.Spec.StorageClasses); i++ {
+		storageClassName := consumerResource.Spec.StorageClasses[i].Name
+
+		provisioner := ""
+		annotations := map[string]string{}
+		labels := map[string]string{}
+		parameters := map[string]string{}
+
+		//TODO: keyRotation and defaultStorageClass annotation
+		if storageClassName == storagecluster.GenerateNameForCephBlockPoolSC(storageCluster) {
```

Comment:
This if-else approach is unreasonable for so many built in *classes. Can we move to a template based approach where the template is chosen based on the name?